cpp/src/fil/infer.cu

```diff
@@ -91,10 +95,40 @@ __device__ __forceinline__ vec<1, output_type> infer_one_tree(tree_type tree,
   return out;
 }
 
+template <int NITEMS>
+struct best_margin_label {
+  vec<NITEMS, float> margin;
+  vec<NITEMS, int> label;
+};
+
+template <int NITEMS>
+struct ArgMax {
+  __device__ best_margin_label<NITEMS> operator()(
```

Comment:
It would also be nice to have this code as a static function named `max()`, since it's also used as a function. Then `ArgMax<NITEMS>::operator()` can simply call `max()` for `best_margin_label<NITEMS>`.